### PR TITLE
Builder API Integration tests

### PR DESCRIPTION
**Pull request by raskchanky**
Test the builder API via the builder dev studio both locally and in travis

Adapted from https://github.com/habitat-sh/habitat/pull/4223

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: test/builder-api/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+habitat/builder-datastore/7043/20180208190943   standalone  up     24          45431  builder-datastore.default   persistent
+habitat/builder-originsrv/7049/20180208225209   standalone  up     19          45827  builder-originsrv.default   persistent
+```
+If we do not build, we'd be testing against the stable package versions of the
+builder components:
+```
+[4][default:/src:0]# build-builder
+```
+```
+[5][default:/src:0]# test/builder-api/test.sh
```
Comment (baumanj) [inline test/builder-api/README.md]: Add `test-builder` helper function

**Comment by raskchanky** [comment]
Awesome job @baumanj!

**Comment by raskchanky** [comment]
tenor-23893831

**Pull request merged**